src/main/java/org/killbill/billing/plugin/adyen/api/ExpiredPaymentPolicy.java

```diff
@@ -81,7 +82,7 @@ private boolean containOnlyAuthsOrPurchases(final List<PaymentTransactionInfoPlu
 
     private DateTime expirationDateForInitialTransactionType(final AdyenPaymentTransactionInfoPlugin transaction) {
         final String paymentMethod = getPaymentMethod(transaction);
-        if (is3ds(transaction)) {
+        if (is3ds(transaction) || isKlarnaPayment(paymentMethod)) {
             return transaction.getCreatedDate().plus(adyenProperties.getPending3DsPaymentExpirationPeriod());
```

Comment:
is the Pending3DsPaymentExpirationPeriod is same as klarnaExpirationPeriod?

Reply:
How is this handled ?
Is this for clean up of Redirectshopper time out responses ?

Reply:
This is where PENDING AUTH transactions will be marked as "CANCELLED" and when Janitor runs (every 5 min), this transaction status will be set to PLUGIN_FAILURE.
org.killbill.billing.plugin.adyen.pending3DsPaymentExpirationPeriod=PT1H